astroid/scoped_nodes.py

```diff
@@ -103,6 +103,39 @@ def _c3_merge(sequences, cls, context):
     return None
 
 
+def clean_typing_generic_mro(sequences: List[List["ClassDef"]]) -> None:
+    """typing.Generic is allowed to appear multiple times in the initial mro.
+    The final one however, MUST only contain ONE.
+
+    This method will check if Generic is in inferred_bases, but also
+    part of bases_mro. If true, remove it from inferred_bases
+    as well as its entry the bases_mro.
+
+    Format sequences: [[self]] + bases_mro + [inferred_bases]
```

Comment:
I wonder if it could not be clearer to pass 3 arguments instead of one (i.e self, bases_mro and inferred_bases).
What do you think about it?